### PR TITLE
Step 1320 ci flakyness

### PR DESCRIPTION
**Pull request by godrei**
### Checklist

- [x] I've read and accepted the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- Requires _PATCH_ [version update](https://semver.org/)

### Context

During the Android Emulator Boot status check, the Step kills adb daemon if the check fails with a timeout error or if the error is `daemon not running; starting now at`.

This PR removes the adb daemon kill command call in case of `daemon not running; starting now at` error, as the error message implies that the daemon is just restarted, it does not make sense to kill it again.

| [From this](https://app.bitrise.io/build/21c86004-67b4-4b0f-a182-21ee88988b4c) | [To this](https://app.bitrise.io/build/f4699e16-4c9d-479c-b8a2-71574e6865fc#?tab=log) |
|-----------|---------|
| Screenshot 2021-08-09 at 10 32 47 | Screenshot 2021-08-09 at 10 32 56 |

### Changes

- Remove adb daemon kill in case of `daemon not running; starting now at` error.

### CODE REVIEW

**Review by Bence1001 — DISMISSED** [review]
LGTM! 👍 

Can we test this behavior somehow?

**Review by mateherber — DISMISSED** [review]

**Comment by godrei** [comment]
@Bence1001 unit tests added in https://github.com/bitrise-steplib/steps-wait-for-android-emulator/pull/19/commits/f95c8f720f3d83fa7a41dc443e0b57669feb6646

**Review by Bence1001 — COMMENTED** [review]
Thanks for adding the tests, good stuff! I have some minor remarks left.

File: clock.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package main
+
+import "time"
+
+// DefaultClock ...
+type DefaultClock struct{}
```
[the remark this answers is not in the document]
Comment (godrei) [reply clock.go]: No, updated in https://github.com/bitrise-steplib/steps-wait-for-android-emulator/pull/19/commits/61295051a07cfe32484514f1741bf80f55383908

File: commands.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+import (
+	"bytes"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+// DefaultCmdRunner ...
+type DefaultCmdRunner struct {
```
[the remark this answers is not in the document]
Comment (godrei) [reply commands.go]: No, updated in https://github.com/bitrise-steplib/steps-wait-for-android-emulator/pull/19/commits/61295051a07cfe32484514f1741bf80f55383908

File: main.go [hunk 3]
```diff
@@ -17,6 +14,23 @@ import (
 	"github.com/bitrise-io/go-utils/log"
 )
 
+// CmdRunner ...
+type CmdRunner interface {
```
[the remark this answers is not in the document]
Comment (godrei) [reply main.go:18]: I do not think so.

While you read the main.go, you can get a high-level overview of the step.
I feel like it is better to separate the implementation details of the used command runner and clock.

[the remark this answers is not in the document]
Comment (Bence1001) [reply main.go:18]: Works for me, thanks!

**Review by Bence1001 — APPROVED** [review]
LGTM! 👍

**Pull request merged**